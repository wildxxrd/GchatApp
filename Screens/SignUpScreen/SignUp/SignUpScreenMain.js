import { StyleSheet, Text, View, TouchableOpacity, Image } from "react-native";
import { firebase } from "../../../config";
import { useState } from "react";
import { TextInput } from "react-native-gesture-handler";

const SignUpScreenMain = ({ navigation }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [lastName, setLastName] = useState("");
  const [firstName, setFirstName] = useState("");

  const registerUser = async (email, password, firstName, lastName) => {
    await firebase
      .auth()
      .createUserWithEmailAndPassword(email, password, firstName, lastName)
      .then(() => {
        firebase
          .auth()
          .currentUser.sendEmailVerification({
            handleCodeInApp: true,
            url: "ggchat-f307f.firebaseapp.com/",
          })
          .then(() => {
            alert("Verification email sent");
          })
          .catch((error) => {
            alert(error.message);
          })
          .then(() => {
            firebase
              .firestore()
              .collection("users")
              .doc(firebase.auth().currentUser.uid)
              .set({
                firstName,
                lastName,
                email,
              });
          })
          .catch((error) => {
            alert(error.message);
          });
      })
      .catch((error) => {
        alert(error.message);
      });
  };

  const signUpTitleText = "Sign Up with Email";
  const registerText = "Already registered? ";

  return (
    <View>
      <Image style={style.img} source={require("../../../assets/logo.jpg")} />
      <Text style={style.titleText}>{signUpTitleText}</Text>
      <View>
        <TextInput
          style={style.input}
          placeholder="First Name"
          placeholderTextColor="black"
          onChangeText={(firstName) => setFirstName(firstName)}
          autoCorrect={false}
        />
        <TextInput
          style={style.input}
          placeholder="Last Name"
          placeholderTextColor="black"
          onChangeText={(lastName) => setLastName(lastName)}
          autoCorrect={false}
        />
        <TextInput
          style={style.input}
          placeholder="Email"
          placeholderTextColor="black"
          onChangeText={(email) => setEmail(email)}
          autoCorrect={false}
          autoCapitalize="none"
          keyboardType="email-address"
        />
        <TextInput
          style={style.input}
          placeholder="Password"
          placeholderTextColor="black"
          onChangeText={(password) => setPassword(password)}
          autoCorrect={false}
          autoCapitalize="none"
          secureTextEntry={true}
        />
      </View>
      <TouchableOpacity
        onPress={() => registerUser(email, password, firstName, lastName)}
      >
        <Text style={style.signUpText}>Sign Up</Text>
      </TouchableOpacity>
      <Text style={style.footerText}>
        {"\n"}
        {registerText}
        <TouchableOpacity onPress={() => navigation.navigate("Sign In")}>
          <Text style={style.loginText}>Login</Text>
        </TouchableOpacity>
      </Text>
    </View>
  );
};

const style = StyleSheet.create({
  img: {
    justifyContent: "center",
    alignItems: "center",
    marginLeft: 60,
    marginBottom: 20,
    width: 250,
    height: 275,
  },
  titleText: {
    fontSize: 30,
    marginHorizontal: 20,
    fontWeight: "bold",
    fontFamily: "Baskerville-SemiBold",
  },
  input: {
    backgroundColor: "#DFDFDE",
    height: 40,
    width: 300,
    margin: 12,
    padding: 10,
    borderRadius: 10,
  },
  signUpText: {
    marginHorizontal: 170,
    color: "#277BC0",
    fontSize: 20,
    fontFamily: "Baskerville-Bold",
  },
  loginText: {
    color: "#277BC0",
    fontSize: 20,
    fontFamily: "Baskerville-Bold",
  },
  footerText: {
    marginHorizontal: 100,
    fontSize: 20,
    fontFamily: "Baskerville",
  },
});

export default SignUpScreenMain;
