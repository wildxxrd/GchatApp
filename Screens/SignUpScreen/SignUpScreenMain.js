import { StyleSheet, Text, View, TextInput, Image } from "react-native";
import SignUpInputField from "./SignUpSignInInputField";
import SignUpButton from "./SignUpButton";
import SignUpLoginButton from "./SignUpLoginButton";

const SignUpScreenMain = () => {
  const signUpTitleText = "Sign Up with Email";
  const registerText = "Already registered? ";

  return (
    <View>
      <Image style={style.img} source={require("../../assets/logo.jpg")} />
      <Text style={style.titleText}>
        {signUpTitleText}
        {"\n"}
        {"\n"}
      </Text>
      <Text style={style.inputText}>{"Full Name"}</Text>
      <SignUpInputField placeHolderText="Enter a name" />
      <Text style={style.inputText}>{"Email Address"}</Text>
      <SignUpInputField placeHolderText="Enter your email address" />
      <Text style={style.inputText}>{"Password"}</Text>
      <SignUpInputField placeHolderText="Enter your password" />
      <SignUpButton />
      <Text style={style.footerText}>
        {"\n"}
        {"\n"}
        {registerText}
        <SignUpLoginButton />
        <Text style={style.loginText}></Text>
      </Text>
    </View>
  );
};

const style = StyleSheet.create({
  img: {
    justifyContent: "center",
    alignItems: "center",
    marginLeft: 30,
    marginBottom: 20,
    width: 250,
    height: 275,
  },
  titleText: {
    fontSize: 30,
    marginHorizontal: 20,
    fontWeight: "bold",
    fontFamily: "Baskerville-SemiBold",
  },
  inputText: {
    fontSize: 16,
    color: "#3D8361",
    fontFamily: "Baskerville-SemiBold",
  },
  loginText: {
    color: "#277BC0",
    fontSize: 16,
    fontFamily: "Baskerville",
  },
  footerText: {
    marginHorizontal: 100,
    fontSize: 16,
    fontFamily: "Baskerville",
  },
});

export default SignUpScreenMain;
