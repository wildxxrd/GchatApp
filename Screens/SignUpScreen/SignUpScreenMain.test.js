import React from 'react';
import renderer from 'react-test-renderer';
import SignUpScreenMain from './SignUpScreenMain';

test('renders the main screen', () => { 
    const signUpScreen = renderer.create(<SignUpScreenMain />);
    expect(signUpScreen).toBeDefined();
 })

 test('Create SnapShot', () => { 
    const signUpScreen = renderer.create(<SignUpScreenMain />);
    expect(signUpScreen).toMatchSnapshot();
 })