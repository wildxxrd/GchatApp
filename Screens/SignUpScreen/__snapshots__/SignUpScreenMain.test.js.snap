// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should first 1`] = `
<View>
  <Image
    source={
      {
        "testUri": "../../../assets/logo.jpg",
      }
    }
    style={
      {
        "alignItems": "center",
        "height": 275,
        "justifyContent": "center",
        "marginBottom": 20,
        "marginLeft": 30,
        "width": 250,
      }
    }
  />
  <Text
    style={
      {
        "fontFamily": "Baskerville-SemiBold",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginHorizontal": 20,
      }
    }
  >
    Sign Up with Email
    

    

  </Text>
  <Text
    style={
      {
        "color": "#3D8361",
        "fontFamily": "Baskerville-SemiBold",
        "fontSize": 16,
      }
    }
  >
    Full Name
  </Text>
  <View>
    <TextInput
      placeholder="Enter a name"
      placeholderTextColor="black"
      style={
        {
          "backgroundColor": "#DFDFDE",
          "borderRadius": 10,
          "height": 40,
          "margin": 12,
          "padding": 10,
          "width": 300,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#3D8361",
        "fontFamily": "Baskerville-SemiBold",
        "fontSize": 16,
      }
    }
  >
    Email Address
  </Text>
  <View>
    <TextInput
      placeholder="Enter your email address"
      placeholderTextColor="black"
      style={
        {
          "backgroundColor": "#DFDFDE",
          "borderRadius": 10,
          "height": 40,
          "margin": 12,
          "padding": 10,
          "width": 300,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#3D8361",
        "fontFamily": "Baskerville-SemiBold",
        "fontSize": 16,
      }
    }
  >
    Password
  </Text>
  <View>
    <TextInput
      placeholder="Enter your password"
      placeholderTextColor="black"
      style={
        {
          "backgroundColor": "#DFDFDE",
          "borderRadius": 10,
          "height": 40,
          "margin": 12,
          "padding": 10,
          "width": 300,
        }
      }
    />
  </View>
  <View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "fontFamily": "Baskerville",
        "fontSize": 16,
        "marginHorizontal": 100,
      }
    }
  >
    

    

    Already registered? 
    <Text
      style={
        {
          "color": "#277BC0",
          "fontFamily": "Baskerville",
          "fontSize": 16,
        }
      }
    >
      Login
    </Text>
  </Text>
</View>
`;
